src/modules/flow/string/string-common.c

```diff
@@ -51,3 +51,226 @@ string_is_empty(struct sol_flow_node *node,
+        return sol_flow_send_string_take_packet(node,
+            SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__STRING, output);
+    } else if (mdata->string_conns == 0 && mdata->blob_conns > 0) {
+        struct sol_blob *blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT,
+            NULL, output, outputlen);
+        SOL_NULL_CHECK_GOTO(blob, error_blob);
+        r = sol_flow_send_blob_packet(node,
+            SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__BLOB, blob);
+        sol_blob_unref(blob);
+        return r;
+    } else {
+        struct sol_blob *blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT,
+            NULL, output, outputlen);
+        int r2;
+        SOL_NULL_CHECK_GOTO(blob, error_blob);
+
+        r = sol_flow_send_string_packet(node,
+            SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__STRING, output);
+
+        r2 = sol_flow_send_blob_packet(node,
+            SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__BLOB, blob);
+        sol_blob_unref(blob);
+
+        return r || r2;
+    }
+
+error:
+    sol_flow_send_error_packet(node, -r,
+        "Could not decode string '%s' to base64: %s",
+        in_value, sol_util_strerrora(-r));
+    sol_buffer_fini(&buf);
+    return r;
+
+error_blob:
+    free(output);
+    return -ENOMEM;
+}
+
+int
+string_b64decode_port_connect(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id)
+{
+    struct string_b64decode_data *mdata = data;
+
+    if (port == SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__STRING)
+        mdata->string_conns++;
+    else if (port == SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__BLOB)
+        mdata->blob_conns++;
+    else
+        return -EINVAL;
+
+    return 0;
+}
+
+int
+string_b64decode_port_disconnect(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id)
+{
+    struct string_b64decode_data *mdata = data;
+
+    if (port == SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__STRING &&
+        mdata->string_conns > 0)
+        mdata->string_conns--;
+    else if (port == SOL_FLOW_NODE_TYPE_STRING_B64DECODE__OUT__BLOB &&
+        mdata->blob_conns > 0)
+        mdata->blob_conns--;
+    else
+        return -EINVAL;
+
+    return 0;
+}
+
+int
+string_b64_open(struct sol_flow_node *node,
+    void *data,
+    const struct sol_flow_node_options *options)
+{
+    struct string_b64_data *mdata = data;
+    const struct sol_flow_node_type_string_b64encode_options *opts;
+
+    SOL_FLOW_NODE_OPTIONS_SUB_API_CHECK
+        (options, SOL_FLOW_NODE_TYPE_STRING_B64ENCODE_OPTIONS_API_VERSION, -EINVAL);
+    /* both b64encode and b64decode have the same options structure, however
+     * the generator emits different symbols. Let's pick one.
+     */
+    opts = (const struct sol_flow_node_type_string_b64encode_options *)options;
+
+    if (!opts->base64_map &&
+        (opts->base64_map[0] == '\0' ||
+        streq(opts->base64_map, SOL_BASE64_MAP)))
+        mdata->base64_map = SOL_BASE64_MAP;
+    else if (opts->base64_map) {
```

Comment:
fair, but will not validate as it's rare to provide those and it's a PITA to check (would need to walk all 65 entries and call `memchr()` to see if it's in the prior or next segment).